### PR TITLE
Fix for get_image_array_size().

### PR DESCRIPTION
**Pull request by scottp101**
processing for get_image_array_size() was falling into the ExtInst case and yielding:

error: 233: Invalid extended instruction number: 138

From the disassembler.

Also, since it returns a size_t, I had to add the return type to the name since, for example, in 64-bit you could have i32 get_image_width() => OpImageQuerySizeLod and i64 get_image_array_size() => OpImageQuerySizeLod which were conflicting.

### CODE REVIEW

File: lib/SPIRV/OCL20ToSPIRV.cpp [hunk 1]
```diff
@@ -855,14 +856,16 @@ OCL20ToSPIRV::visitCallGetImageSize(CallInst* CI,
       assert(IsImg);
       Desc = map<SPIRVTypeImageDescriptor>(TyName.str());
       Dim = getImageDimension(Desc.Dim) + Desc.Arrayed;
-      Ret = Type::getInt32Ty(*Ctx);
+      Ret = CI->getCalledFunction()->getReturnType();
       if (Dim > 1)
         Ret = VectorType::get(Ret, Dim);
       if (Desc.Dim == DimBuffer)
-        return getSPIRVFuncName(OpImageQuerySize);
+          return getSPIRVFuncName(OpImageQuerySize,
```
Comment (yxsamliu) [inline lib/SPIRV/OCL20ToSPIRV.cpp]: Can we define a function
std::string getSPIRVFuncName(Op OC, Type *RetTy);
and use that?

**Comment by yxsamliu** [comment]
Does this pass the lit test OpImageQuerySize.ll?

File: lib/SPIRV/OCL20ToSPIRV.cpp (hunk 1 above)
Comment (scottp101) [inline lib/SPIRV/OCL20ToSPIRV.cpp]: Sure, Latest change has it.

**Comment by scottp101** [comment]
Passing with latest change.

File: lib/SPIRV/SPIRVUtil.cpp [hunk 2]
```diff
@@ -363,6 +363,11 @@ getSPIRVFuncName(Op OC, StringRef PostFix) {
 }
 
 std::string
+getSPIRVFuncName(Op OC, const Type *pRetTy, bool IsSigned) {
+  return prefixSPIRVName(getName(OC) + kSPIRVPostfix::Divider + getPostfixForReturnType(pRetTy, false));
```
Comment (yxsamliu) [inline lib/SPIRV/SPIRVUtil.cpp]: Better floow LLVM coding standard http://llvm.org/docs/CodingStandards.html
line width should be < 80

**Comment by scottp101** [comment]
Sure thing.

**Comment by AndreyE** [comment]
Hi, guys,
Do those fixes follow the 'round trip' conversion agreement? It seems these commits are mostly to the writerr.
Could you fix the reader too, please?

**Pull request merged**